sql/memory.go

```diff
@@ -60,10 +60,20 @@ type RowsCache interface {
 	// free some memory. If after that there is still no memory available, it
 	// will return an error and erase all the content of the cache.
 	Add(Row) error
-	// Get all rows.
+	// Get returns all rows.
 	Get() []Row
 }
 
+// RowsCache is a cache of rows.
+type Rows2Cache interface {
+	// Add2 a new row to the cache. If there is no memory available, it will try to
```

Comment:
we should embed RowsCache here if we're going to do it for other core interfaces